framework/http/MessageTrait.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\http;
+
+use Psr\Http\Message\StreamInterface;
+use yii\di\Instance;
+
+/**
+ * MessageTrait provides set of methods to satisfy [[\Psr\Http\Message\MessageInterface]].
+ *
+ * This trait should be applied to descendant of [[\yii\base\BaseObject]] implementing [[\Psr\Http\Message\MessageInterface]].
```

Comment:
Why `descendant of [[\yii\base\BaseObject]]` is a requirement here?

Reply:
Otherwise virtual properties declared in trait's PHPDoc will not appear.